tests/check_parachain_lease_expiration.ts

```diff
@@ -0,0 +1,78 @@
+import { ApiPromise, WsProvider } from '@polkadot/api';
+import '@polkadot/api-augment';
+import '@polkadot/api-augment/kusama';
+import '@polkadot/api-augment/polkadot';
+import { assert } from 'chai';
+
+export async function createPromiseApi(nodeAddress: string) {
+  const wsProvider = new WsProvider(nodeAddress);
+
+  const api = new ApiPromise({
+    provider: wsProvider,
+  });
+  await api.isReady;
+  console.log(`${nodeAddress} has been started`);
+  return api;
+}
+
+describe('Check Parachain_Lease_Expiration', () => {
+    it('Check Manta Parachain Lease', async () => {
+        const polkadotNode = 'wss://1rpc.io/dot';
+        const polkadotApi = await createPromiseApi(polkadotNode);
+        const mantaParaId = 2104;
+        const oneDay = 3600 * 24;
+
+        const relaychainBlockTime = polkadotApi.consts.babe.expectedBlockTime.toNumber() / 1000;
+        const bestNumber = (await polkadotApi.derive.chain.bestNumber()).toNumber();
+        const startNumber = bestNumber - polkadotApi.consts.slots.leaseOffset.toNumber();
+        const leasePeriod = polkadotApi.consts.slots.leasePeriod.toNumber();
+        const currentPeriod = Math.floor(startNumber / leasePeriod);
+
+        const remainingLeaseLength = await polkadotApi.query.slots.leases(mantaParaId);
+        const endPeriod = currentPeriod + remainingLeaseLength.length - 1;
+
+        await polkadotApi.disconnect();
+        assert(remainingLeaseLength.length >= 1, 'Manta lease is going to expire, please do auction.');
+        console.log(`Remaining lease period for manta: ${currentPeriod}-${endPeriod}`);
+
+        const firstBlock = startNumber % leasePeriod;
+        if (firstBlock === 0) {
+            const remainingDays = leasePeriod * relaychainBlockTime * remainingLeaseLength.length / oneDay;
+            console.log(`Manta parachain lease has ${remainingDays} days remaining.`);
+        } else {
+            const currentLeaseRemainedBlocks = leasePeriod - firstBlock;
+            const remainingDays = leasePeriod * relaychainBlockTime * (remainingLeaseLength.length - 1) / oneDay + (currentLeaseRemainedBlocks * relaychainBlockTime / oneDay);
+            console.log(`Manta parachain lease has ${remainingDays} days remaining.`);
+        }
+    }).timeout(2000000);
+
+    it.skip('Check Calamari Parachain Lease', async () => {
```

Comment:
This skips calamari test? I suppose it would fail right now so it needs to be activated when calamari is back online?

Reply:
Yes, it will fail. I have an issue #1139  to track it, once we get the slot again, this test case will be enabled.

Reply:
i dont get how this fails at all, the only output i see is a `console.log`

Reply:
It has `it.skip(` to avoid failure right now. 

Will need to be update when we get calamari a slot again

Reply:
can you explain how it will fail? there's no return value

Reply:
Line 65

```ts
 assert(remainingLeaseLength.length >= 1, 'Calamari lease is going to expire, please do auction.');
```